app/models/audit/display_item.rb

```diff
@@ -68,22 +108,13 @@ def compute_entity_name(klass)
   end
 
   def compute_entity_display_name(klass)
-    # Try to get the actual record and use its name method first
-    if klass
-      begin
-        record = klass.with_deleted.find_by(id: version.item_id)
-        if record.respond_to?(:entity_name)
-          return record.entity_name
-        elsif record.respond_to?(:name)
-          return record.name
-        end
-      rescue StandardError
-        # If we can't load the record, fall back to humanized entity type
-      end
-    end
+    record = @records_by_type_and_id[version.item_type][version.item_id]
```

Comment:
Is there ever a situation where `@records_by_type_and_id[version.item_type]` would return `nil`, if so, we should probably rewrite with tries